src/map/entities/battleentity.cpp

```diff
@@ -313,6 +313,12 @@ int16 CBattleEntity::GetWeaponDelay(bool tp)
     TracyZoneScoped;
     if (StatusEffectContainer->HasStatusEffect(EFFECT_HUNDRED_FISTS) && !tp)
     {
+        if (this->objtype == ENTITYTYPE::TYPE_MOB)
+        {
+            // Captures show mobs swing around 700 delay under hundered fists
```

Comment:
Hundred fists doesn't actually set a specific delay value, it causes a reduction that some sources say is 75% and other say is 80%

To the best of my knowledge this is true for both players and mobs, the difference being mobs don't have added delay from a h2h weapon and are -usually- using the basic unarmed h2h delay as their basis (there exist weird cases like non monk non h2h mobs using it). So in theory, whatever delay the mob already had looked like the values seen after hundred fists delay reduction applied. But static setting a value is going to give edge cases down the road even though you are improving it over the previous wrong static value, and so it might be better to either work out the correct math for the reduction -or- prove and document mobs work differently right now to justify the static value once and for all.

...Also the fact players are getting a static value when the reduction is pretty well known kills me.

Reply:
## Note, these quotes are for players (highlighting that the 1700 isn't correct for players either)

old clopedia wiki : 
> Upon usage, character's melee attack delay is greatly reduced, similar to a massive Haste effect. The effect does not kick in until the next melee strike is made, as attack delay is calculated immediately following each attack round.
> - Delay is not reduced to zero, or any other set amount. Attack rate is still based on the delay of your weapon. Higher delay weapons will still attack slower than lower delay weapons will.
>
> Gives the effect of 80% haste, not affected by magical hastes (i.e. marches, haste spell) and also not affected by gear haste.
> - During Hundred Fists, it is recommended to replace Haste gear with Accuracy / Attack / STR gear, to maximize damage output.
> - Completely unaffected by all forms of slow, including raise weakness

BG wiki :
> Reduces delay by 75% after calculated delay
> - Calculated delay is delay after including Weapon Delay and Martial Arts traits and gear.
> - Because your post-Arts delay is used, it is possible to break the normal 80% [Attack Speed](https://www.bg-wiki.com/ffxi/Attack_Speed) cap with this ability when using Hand-to-Hand weapons.

wiki.ffo.jp :
> マーシャルアーツ による隔の減算をした後の攻撃間隔が1/4になる。この時の攻撃間隔]はヘイストキャップ80%を超えて短縮される
マーシャルアーツ効果アップなどのような得TP減少は無い（隔マイナスではない）
全てのヘイスト・スロウ効果が無視される

(translated) :
> After subtracting the distance from martial arts, the attack interval becomes 1/4. At this time, the attack interval] will be reduced by more than 80% of the haste cap.
There is no reduction in TP gain like when martial arts effects are increased (not a negative difference)
All haste and slow effects are ignored.

Reply:
PS, pretty sure this is milliseconds and not actual delay, because 1700 delay is insanely long 

1700 ms is massive too, but gets divided to something more reasonable

I only mention this because the comment says "700 delay". Even 700 delay would be huge.  You may want to amend that to "about 700ms".

Really kills me that the core has multiple cases of using the word delay without obvious distinguishing between weapon delay units and milliseconds. (see issue 2502...)

Reply:
> Hundred fists doesn't actually set a specific delay value, it causes a reduction that some sources say is 75% and other say is 80%
> 
> To the best of my knowledge this is true for both players and mobs, the difference being mobs don't have added delay from a h2h weapon and are -usually- using the basic unarmed h2h delay as their basis (there exist weird cases like non monk non h2h mobs using it). So in theory, whatever delay the mob already had looked like the values seen after hundred fists delay reduction applied. But static setting a value is going to give edge cases down the road even though you are improving it over the previous wrong static value, and so it might be better to either work out the correct math for the reduction -or- prove and document mobs work differently right now to justify the static value once and for all.
> 
> ...Also the fact players are getting a static value when the reduction is pretty well known kills me.

Yeah the mob captures suggest that the HF value is either a fixed delay of 85 or if you assume the base delay of monk mobs is 480 (similar to players) then 85 would be about an 80% reduction in delay. 

The captures also suggest that (at least for mobs) the base delay used does not take martial arts into account. For example Hippolytos (with MA3) showed pre-HF delay of 364 and a HF delay of  84. While Tieholtsodi (with MA5-6) has pre-HF delay of 213 and also a HF-delay of 85.

I would be ok with using an 80% delay reduction from the base delay for mobs instead of fixed value.

Reply:
> The captures also suggest that (at least for mobs) the base delay used does not take martial arts into account.

thats how it is for players 2 - all other forms of delay reduction get ignored, hf becomes the only reduction in play, but its massive

Reply:
Youtube videos are practically useless for determining delays. Even if I trusted doing some kind of frame counting or clock watching and assumed nothing happened to the playback rate or frame rate, all we'd get is an approximate time between hits and not how that time was determined. 

I do not doubt that the 700ms was about or even exactly the time between punches here in the examples the PR author was working with, we're discussing the methods it got to that value from the normal delay before hundred fists was activated.

Reply:
@TracentEden 
> I would be ok with using an 80% delay reduction from the base delay for mobs instead of fixed value.

I would be ok merging that, for both players and monsters.

Reply:
> @TracentEden
> 
> > I would be ok with using an 80% delay reduction from the base delay for mobs instead of fixed value.
> 
> I would be ok merging that, for both players and monsters.

Yeah, I am asking capture person to check about martial arts and player HF (as that is mentioned in the wikis you linked), thus can make sure both player and mob logic is correct.